main/src/ca/uwaterloo/flix/language/ast/OccurrenceAst.scala

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright 2024 Jakob Schneider Villumsen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.ast
+
+import ca.uwaterloo.flix.language.ast.shared.SymUse.{CaseSymUse, EffectSymUse, OpSymUse}
+import ca.uwaterloo.flix.language.ast.shared.*
+import ca.uwaterloo.flix.util.collection.Nel
+
+object OccurrenceAst {
+
+  case class Root(defs: Map[Symbol.DefnSym, Def],
+                  enums: Map[Symbol.EnumSym, MonoAst.Enum],
+                  structs: Map[Symbol.StructSym, MonoAst.Struct],
+                  effects: Map[Symbol.EffectSym, MonoAst.Effect],
+                  mainEntryPoint: Option[Symbol.DefnSym],
+                  entryPoints: Set[Symbol.DefnSym],
+                  sources: Map[Source, SourceLocation])
+
+  case class Def(sym: Symbol.DefnSym, fparams: List[(MonoAst.FormalParam, Occur)], spec: MonoAst.Spec, exp: Expr, context: DefContext, loc: SourceLocation)
```

Comment:
```
fparams: List[(MonoAst.FormalParam, Occur)]
```

I think this is "laziness" left over from last implementation.

Why not add our own `OccurrenceAst.FormalParam` and then it can also have the `Occur` information.

That way we avoid awkward tuples.